Gemfile

```diff
@@ -10,6 +10,13 @@ gem "chef-config", git: "https://github.com/chef/chef", branch: "main", glob: "c
 gem "chef-utils", git: "https://github.com/chef/chef", branch: "main", glob: "chef-utils/chef-utils.gemspec"
 # NOTE: do not submit PRs to add pry as a dep, add to your Gemfile.local
 group :development do
+  install_if -> { Gem::Version.new("3.4").to_s >= (RUBY_VERSION).to_s } do
+    gem "racc", "~> 1.8.0"
+    gem "base64"
+  end
+  install_if -> { Gem::Version.new("3.3").to_s >= (RUBY_VERSION).to_s } do
+    gem "racc", "~> 1.8.0"
```

Comment:
@johnmccrae this conflicts with the above `racc` specification for 3.4. The `install_if` ultimately acts as though the `gem` command were evaluated at the top or group level for the purpose of recording the `Gemfile.lock`.

Also the version should be the following (the `.to_s` is unnecessary and breaks the comparison logic in certain scenarios such as (because string comparison instead of version comparison)
```ruby
irb(main):011> Gem::Version.new("3.3").to_s >= "3.3.0".to_s
=> false
```

This should be the order of things (validated that `RUBY_VERSION` first works)
```ruby
  install_if -> { RUBY_VERSION >= Gem::Version.new("3.4") } do
    gem "base64"
  end
  install_if -> { RUBY_VERSION >= Gem::Version.new("3.3") } do
    gem "racc", "~> 1.8.0"
  end
  ```

Reply:
Also, there should be `Gemfile.lock` output that has `racc` and `base64`

Reply:
FWIW, I found `install_if` to be super unreliable and switched to simple `if`s which worked much better.